src/components/TreeNode.tsx

```diff
@@ -81,14 +84,39 @@ const TreeNode = ({
     return lastPatches?.some(patch => deleteActions.includes(patch.actions[0]))
   })
 
+  // true if the last action is swapParent
+  const isLastActionSwapParent = useSelector(state => {
+    const lastPatches = state.undoPatches[state.undoPatches.length - 1]
+    return lastPatches?.some(patch => patch.actions[0] === 'swapParent')
+  })
+
+  const isLastActionMoveThought = useSelector(state => {
+    const lastPatches = state.undoPatches[state.undoPatches.length - 1]
+    return lastPatches?.some(patch => patch.actions[0] === 'moveThought')
+  })
```

Comment:
You're probably going to run into trouble with this. Any action can come before the swap, including actions that use `moveThought` internally. For example, `categorize` uses `moveThought` internally, but its action type is `categorize`. Additionally, `undo/redo` breaks the rule, as the temporally last action may come from `undoPatches` or `redoPatches`.

Basically, any condition that checks for a specific action before the swap is going to have breaking cases. A swap may be preceded by *any* action. We cannot rely on a specific preceding action.

Reply:
I'm still working on this. I'm now approaching it from a different angle to see if the dynamic top positions are what's causing the unexpected trailing animation. The goal is to avoid relying on any conditions based on undoPatches.

Reply:
I've fixed this by setting the `top` position of the outer div to `auto` by default. The root cause was that it had been set to 0, which caused the animation to start at the top of the viewport, resulting in an unexpected trailing transition.

The final result now looks like this:

https://github.com/user-attachments/assets/e77bcfa1-8a8b-4672-8212-8f86ae277bf1